std/conv.d

```diff
@@ -2541,34 +2541,48 @@ Target parse(Target, Source)(ref Source p)
         return new ConvException(text(msg, " for input \"", p, "\"."), fn, ln);
     }
 
-    enforce(!p.empty, bailOut());
+    static if (isNarrowString!Source)
+    {
+        import std.string : representation, assumeUTF;
+        auto source = p.representation;
+    }
+    else
+    {
+        alias source = p;
+    }
+
+    enforce(!source.empty, bailOut());
 
-    char sign = 0;                       /* indicating +                 */
-    switch (p.front)
+    bool sign = false;
+    switch (source.front)
     {
     case '-':
-        sign++;
-        p.popFront();
-        enforce(!p.empty, bailOut());
-        if (toLower(p.front) == 'i')
+        sign = true;
+        source.popFront();
```

Comment:
(For simpler refactoring: continue using `p` and give the initial parameter a different name.)